File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        int i = 0, j = 0;
        bool bFlag = false;

        for ( i = 0; i < nums.size(); i++)
        {
            for ( j = i + 1; j < nums.size(); j++)
            {
                if (nums[i] == nums[j])
                {
                    bFlag = true;
                }
            }
        }
        return bFlag;        
    }
};

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        last_seen = {}  # Maps: number -> its most recent index
        
        for i, num in enumerate(nums):
            # If we've seen this number before AND it's close enough, we're done!
            if num in last_seen and i - last_seen[num] <= k:
                return True
            
            # Update the map with the current index
            last_seen[num] = i
            
        return False